File: app/src/main/java/com/novel/read/base/NovelBaseFragment.java
package com.novel.read.base;


import android.content.Intent;
import android.os.Bundle;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import butterknife.ButterKnife;
import butterknife.Unbinder;


public abstract class NovelBaseFragment extends Fragment {

    private Unbinder bind;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View mContextView = inflater.inflate(getLayoutId(), container, false);
        bind = ButterKnife.bind(this, mContextView);
        initView();
        initData();
        return mContextView;
    }


    protected abstract int getLayoutId();

    protected abstract void initView();

    protected abstract void initData();

    public void toActivity(Class<?> cls) {
        Intent intent = new Intent(getActivity(), cls);
        startActivity(intent);
    }

    public void toActivity(Class<?> toClsActivity, Bundle bundle) {
        Intent intent = new Intent(getActivity(), toClsActivity);
        if (bundle != null) {
            intent.putExtras(bundle);
        }
        startActivity(intent);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        bind.unbind();
    }
}

File: app/src/main/java/com/novel/read/base/NovelBaseFragment.kt
package com.novel.read.base


import android.content.Intent
import android.os.Bundle

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment


abstract class NovelBaseFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(getLayoutId(),null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        initData()
    }

    /**
     * 加载布局
     */
    @LayoutRes
    abstract fun getLayoutId():Int

    abstract fun initView()

    abstract fun initData()

    fun toActivity(toClsActivity: Class<*>) {
        this.toActivity(toClsActivity, null as Bundle?)
    }

    fun toActivity(toClsActivity: Class<*>, bundle: Bundle?) {
        val intent = Intent(this.context, toClsActivity)
        if (bundle != null) {
            intent.putExtras(bundle)
        }

        this.startActivity(intent)
    }

    fun toActivityForResult(toClsActivity: Class<*>, bundle: Bundle?, requestCode: Int) {
        val intent = Intent(this.context, toClsActivity)
        if (bundle != null) {
            intent.putExtras(bundle)
        }

        startActivityForResult(intent, requestCode)
    }


}
